qiskit_experiments/analysis/data_processing.py

```diff
@@ -0,0 +1,104 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Curve fitting xydata for experiment analysis
+"""
+# pylint: disable = invalid-name
+
+from typing import List, Dict, Tuple
+import numpy as np
+
+
+def filter_data(data: List[Dict[str, any]], **filters) -> List[Dict[str, any]]:
+    """Return the list of filtered data
+
+    Args:
+        data: list of data dicts.
+        filters: kwargs for filtering based on metadata
+                 values.
+
+    Returns:
+        The list of filtered data. If no filters are provided this will be the
+        input list.
+
+    .. note::
+        The filtered data list does not make a copy of the items in the input
+        data last.
+    """
+    if not filters:
+        return data
+    filtered_data = []
+    for datum in data:
+        include = True
+        metadata = datum["metadata"]
+        for key, val in filters.items():
+            if key not in metadata or metadata[key] != val:
+                include = False
+                break
+        if include:
+            filtered_data.append(datum)
+    return filtered_data
+
+
+def mean_xy_data(xdata: np.ndarray, ydata: np.ndarray) -> Tuple[np.ndarray]:
```

Comment:
`curve_fit_data` also returns a sigma. Should this use the sigma to weight the mean?

Reply:
Yes that would be nice, but I wasn't sure of the correct way to do this though...